daps-xslt/migrate/suse-upgrade.xsl

```diff
@@ -780,6 +803,25 @@
     </xsl:copy>
   </xsl:template>
 
+  <xsl:template match="link[@linkend]" priority="200">
```

Comment:
Since this template uses `link` -- is this running after everything has been converted from DB4 to DB5 already?

Reply:
> is this running _after_ everything has been converted from DB4 to DB5 already?

No, this is the template that is used to convert from DB4 -> DB5.

Perhaps I should better use a different stylesheet (the once from DB5 -> Geekodoc?) and leave this one untouched. It's kind of work in progress, but actually it serves as a backup just in case. :wink:

Reply:
No, all I meant to say is that you probably need to match `ulink[linkend]` and not `link[linkend]` because DB4 does not know the link element afaik (and correct me if I am wrong about that).

Reply:
I see what you mean. Actually the rule is correct as strange as it sounds. I was also surprised, but it's a valid DocBook 4 element (see https://tdg.docbook.org/tdg/4.5/link.html).

For example, in the file [stor_admin.xml](https://github.com/SUSE/doc-sle-legacy/blob/maintenance/SLE11SP4/xml/stor_admin.xml#L1513) you can see a `<link linkend="sec-inst-system-iscsi-target-yast">` tag.

That's why I've integrated this rule.

Reply:
Ah! I did miss that information. Resolving this discussion then.